snapshot-replicator/main.tf

```diff
@@ -242,49 +235,168 @@ resource "aws_db_event_subscription" "default" {
+  principal     = "sns.amazonaws.com"
+  source_arn    = aws_sns_topic.rds_backup_events[0].arn
 }
 
 #Creation of cronjobs
 #Job for triggering backups
-resource "aws_cloudwatch_event_rule" "every_6_hours" {
-  count = var.enable ? 1 : 0
-  name = "every-6-hours-${var.environment}"
-  description = "Fires every 6 hours"
-  schedule_expression = "rate(6 hours)"
+resource "aws_cloudwatch_event_rule" "invoke_rds_snapshot_lambda" {
+  count               = var.enable ? 1 : 0
+  name                = "invokes-rds-snapshot-lambda-${var.environment}"
+  description         = "Fires every 6 hours"
+  schedule_expression = "rate(${var.custom_snapshot_rate} hours)"
 }
 
 #Job for cleaning up old retention
-resource "aws_cloudwatch_event_rule" "every_24_hours" {
-  count = var.enable ? 1 : 0
-  name = "every-24-hours-${var.environment}"
-  description = "Fires every 24 hours"
+resource "aws_cloudwatch_event_rule" "invoke_rds_cleanup_lambda" {
+  count               = var.enable ? 1 : 0
+  name                = "invoke-rds-cleanup-lambda-${var.environment}"
+  description         = "Fires every 24 hours"
   schedule_expression = "rate(24 hours)"
 }
 
 resource "aws_cloudwatch_event_target" "activate_lambda_cron" {
-  count = var.enable ? 1 : 0
-  rule = aws_cloudwatch_event_rule.every_6_hours[0].name
+  count     = var.enable ? 1 : 0
+  rule      = aws_cloudwatch_event_rule.invoke_rds_snapshot_lambda[0].name
   target_id = "rds-backup"
-  arn = aws_lambda_function.rds_create_snapshot[0].arn
+  arn       = aws_lambda_function.rds_snapshot_create[0].arn
 }
 
 resource "aws_cloudwatch_event_target" "activate_lambda_removal_cron" {
-  count = var.enable ? 1 : 0
-  rule = aws_cloudwatch_event_rule.every_24_hours[0].name
+  count     = var.enable ? 1 : 0
+  rule      = aws_cloudwatch_event_rule.invoke_rds_cleanup_lambda[0].name
   target_id = "rds-retention-removal"
-  arn = aws_lambda_function.remove_snapshot_retention[0].arn
+  arn       = aws_lambda_function.rds_snapshot_cleanup[0].arn
+}
+
+
+resource "aws_lambda_permission" "cloudwatch_invoke_rds_snapshot_lambda" {
+  count         = var.enable ? 1 : 0
+  statement_id  = "AllowExecutionFromCloudWatch"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.rds_snapshot_cleanup[0].function_name
+  principal     = "events.amazonaws.com"
+  source_arn    = aws_cloudwatch_event_rule.invoke_rds_snapshot_lambda[0].arn
+}
+
+resource "aws_lambda_permission" "cloudwatch_invoke_rds_cleanup_lambda" {
+  count         = var.enable ? 1 : 0
+  statement_id  = "AllowExecutionFromCloudWatch"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.rds_snapshot_cleanup[0].function_name
+  principal     = "events.amazonaws.com"
+  source_arn    = aws_cloudwatch_event_rule.invoke_rds_cleanup_lambda[0].arn
+}
+
+
+
+# Cleanup of backups on remote region
+resource "aws_lambda_function" "rds_snapshot_cleanup_remote" {
```

Comment:
You should also attach the `AWSLambdaBasicExecutionRole` policy to all lambda execution roles, so they're able to log to Cloudwatch logs